homeassistant/components/renault/renault_entities.py

```diff
@@ -0,0 +1,131 @@
+"""Base classes for Renault entities."""
+from __future__ import annotations
+
+from typing import Any, cast
+
+from renault_api.kamereon.enums import ChargeState, PlugState
+from renault_api.kamereon.models import (
+    KamereonVehicleBatteryStatusData,
+    KamereonVehicleChargeModeData,
+    KamereonVehicleCockpitData,
+    KamereonVehicleHvacStatusData,
+)
+
+from homeassistant.helpers.entity import DeviceInfo, Entity
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import slugify
+
+from .renault_vehicle import RenaultVehicleProxy
+
+ATTR_LAST_UPDATE = "last_update"
+
+
+class RenaultDataEntity(CoordinatorEntity, Entity):
+    """Implementation of a Renault entity with a data coordinator."""
+
+    def __init__(
+        self, vehicle: RenaultVehicleProxy, entity_type: str, coordinator_key: str
+    ) -> None:
+        """Initialise entity."""
+        super().__init__(vehicle.coordinators[coordinator_key])
+        self.vehicle = vehicle
+        self._entity_type = entity_type
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return a device description for device registry."""
+        return self.vehicle.device_info
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique identifier for this entity."""
+        return slugify(f"{self.vehicle.details.vin}-{self._entity_type}")
```

Comment:
why does this need to be slugified?

Reply:
This is only to make it more standard in the database:
- `VF1ABCDE012345678-Mileage` => `vf1abcde012345678_mileage`
- `VF1ABCDE012345678-Outside Temperature` => `vf1abcde012345678_outside_temperature`